.github/workflows/image.yaml

```diff
@@ -0,0 +1,115 @@
+on:
+  push:
+    branches:
+      - "main"
+    tags:
+      - "v*.*.*"
+
+name: Build Docker images
+
+permissions:
+  pull-requests: read
+  contents: read
+  packages: write
+
+jobs:
+  build_dev_image:
+    name: Build development image
+    if: github.ref_type == 'branch' && github.ref == 'refs/heads/main'
+    runs-on: ubuntu-latest
+    environment: dev
+
+    steps:
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - uses: webfactory/ssh-agent@v0.8.0
+        with:
+          ssh-private-key: ${{ secrets.SSH_KEY }}
+
+      - name: Docker metadata
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            {{ github.repository_owner }}/twake-web
+            ghcr.io/${{ github.repository_owner }}/twake-web
+          tags: |
+            type=ref,event=branch
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push image
+        uses: docker/build-push-action@v4
+        with:
+          push: true
+          cache-from: |
+            type=gha
+          cache-to: |
+            type=gha
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          ssh: |
+            default
+
+  build_release_image:
+    name: Build release image
+    if: github.ref_type == 'tag' && startsWith(github.ref, 'refs/tags/')
+    runs-on: ubuntu-latest
+    environment: prod
+
+    steps:
+      - name: Set up Docker Buildx
```

Comment:
duplicate steps
I think we should reuse jobs
https://docs.github.com/en/actions/using-workflows/reusing-workflows

Reply:
No, this is for reusing whole jobs instead of just steps.